openmc/mgxs/library.py

```diff
@@ -717,3 +725,307 @@ def load_from_file(filename='mgxs', directory='mgxs'):
 
         # Load and return pickled Library object
         return pickle.load(open(full_filename, 'rb'))
+
+    def write_mg_library(self, xs_type='macro', domain_names=None, xs_ids=None,
+                         filename='mg_cross_sections', directory='./',
+                         return_names=True):
+        """Creates a cross-section data library file for the Multi-Group
+        mode of OpenMC.
+
+        Parameters
+        ----------
+        xs_type: {'macro', 'micro'}
+            Provide the macro or micro cross section in units of cm^-1 or
+            barns. Defaults to 'macro'. If the Library object is not tallied by
+            nuclide this will be set to 'macro' regardless.
+        domain_names : Iterable of str
+            List of names to apply to the xsdata entries in the
+            resultant mgxs data file. Defaults to "set1", "set2", ...
+        xs_ids : str or Iterable of str
+            Cross section set identifier (i.e., "71c") for all
+            data sets (if only str) or for each individual one
+            (if iterable of str). Defaults to '1g'
+        filename : str
+            Filename for the pickle file. Defaults to 'mg_cross_sections'.
+        directory : str
+            Directory for the pickle file. Defaults to './' (the
+            current working directory).
+        return_names : bool
+            Flag to indicate if the user would like the names of the
+            materials generated by this function returned with completion.
+            Defaults to True.
+
+        Returns
+        -------
+        mat_names : Iterable of str
+            Iterable of material names generated during this routine and
+            applies to the cross section library.  Note this is returned if
+            the return_names parameter is provided.
+
+        Raises
+        ------
+        ValueError
+            When the Library object is initialized with insufficient types of
+            cross sections for the Library.
+
+        See also
+        --------
+        Library.dump_to_file(mgxs_lib, filename, directory)
+
+        """
+
+        # Check the provided parameters
+        cv.check_value('xs_type', xs_type, ['macro', 'micro'])
```

Comment:
Is it true that this method will only work if tally data has been loaded into the `MGXS` objects? If so, this method should error out if a statepoint has not yet been loaded.